sys/pm_layered/pm.c

```diff
@@ -36,11 +36,13 @@
 /**
  * @brief Global variable for keeping track of blocked modes
  */
-static volatile pm_blocker_t pm_blocker = { .val_u32 = PM_BLOCKER_INITIAL };
+static pm_blocker_t pm_blocker = { .val_u32 = PM_BLOCKER_INITIAL };
```

Comment:
Can't we just use `atomic_uint_fast32_t` for `.val_u32` instead?

Reply:
In theory no, because the memory layout of `atomic_uint_fast32_t` could be different to the one of `uint32_t`. (E.g. if it is implemented using a mutex.) In practice, RIOT disables IRQs while accessing atomic variables for which no native atomic instruction sequence can be generated by the compiler, so they would overlap. I still have a bad feeling with atomics in a `union`...

Also: For multiple subsequent accesses to atomic variables disabling IRQs only once before the first access and re-enabling only once after the last is faster and uses less ROM. E.g. `pm_set_lowest()` as up to four accesses to `pm_blocker` in a row, in which case simply disabling IRQs is easier.